packages/core/test/invoke.test.ts

```diff
@@ -2709,6 +2709,31 @@ describe('invoke', () => {
       interpret(machine).start();
     });
   });
+
+  it('invoke creator should provide reference to self', (done) => {
+    const machine = createMachine({
+      invoke: {
+        src: (_ctx, _e, { self }) => () => {
+          self.send('FROM_INVOKED');
```

Comment:
It would be great if we could have some type tests that would verify that only accepted events can be sent to `self`. This test types `self` as:
```ts
ActorRef<AnyEventObject, State<unknown, AnyEventObject, any, {
    value: any;
    context: unknown;
}>>
```
so there is no such validation happening here. Any event gets accepted.

Also - the `stop` method is still a part of `self` (but it's optional). I wonder if we want to hide it completely in the future?

Reply:
> It would be great if we could have some type tests that would verify that only accepted events can be sent to self.

This is recently added to 934fa862c5cc13bcc01fec829309e9e4a184ee3a

> Also - the stop method is still a part of self (but it's optional). I wonder if we want to hide it completely in the future?

That makes sense - will do in a separate PR for v5.